dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -590,6 +590,60 @@ public Result queryUserList(User loginUser, String searchVal, Integer pageNo, In
         return result;
     }
 
+    /**
+     * grant project by code
+     *
+     * @param loginUser login user
+     * @param userId user id
+     * @param projectCodes project code array
+     * @return grant result code
+     */
+    @Override
+    @Transactional(rollbackFor = RuntimeException.class)
```

Comment:
There should be exceptions displayed in the code, otherwise `Transactional` will not work